pkg/visor/visor.go

```diff
@@ -282,6 +288,75 @@ func (node *Node) Start() error {
 	return nil
 }
 
+// RunDaemon starts the auto-peering-daemon as an external process
+func (node *Node) RunDaemon() {
+	bin, err := exec.LookPath("daemon")
+	if err != nil {
+		node.logger.Fatalf("Cannot find `skywire-peering-daemon` binary in $PATH: %s", err)
+	}
+
+	dir, err := ioutil.TempDir("", "named_pipes")
+	if err != nil {
+		node.logger.Errorf("Couldn't create named_pipes dir: %s", err)
+	}
+
+	namedPipe := filepath.Join(dir, "stdout")
+	pubKey := node.conf.Node.StaticPubKey.Hex()
+	err = syscall.Mkfifo(namedPipe, 0600)
+	if err != nil {
+		node.logger.Fatal(err)
+	}
+
+	if err := execute(bin, pubKey, namedPipe); err != nil {
+		node.logger.Fatal("Failed to start daemon as an external process: ", err)
+	}
+
+	node.logger.Info("Opening named pipe for reading packets from skywire-peering-daemon")
+	stdOut, err := os.OpenFile(namedPipe, os.O_RDONLY, 0600)
+	if err != nil {
+		node.logger.Fatal(err)
+	}
+
+	c := make(chan notify.EventInfo, 5)
+	notify.Watch(namedPipe, c, notify.Write)
+
+	go func() {
+		// Read packets from named pipe
+		for {
+			var (
+				packet apd.Packet
+				buff   bytes.Buffer
+			)
+
+			<-c
+			_, err = io.Copy(&buff, stdOut)
+			if err != nil {
+				node.logger.Fatal(err)
```

Comment:
Is it intended behavior that `visor` fails if `daemon` fails to start?

Reply:
Not at all. A daemon failing shouldn't also fail the visor, rather the daemon process should be stopped.

Reply:
@Kifen `Fatal()` calls `os.Exit(1)` and exit the process. So I think we should use `Error()` with `return`.

Reply:
@Kifen I would also unlock the mutex with `defer` right after locking to ensure it's unlocked under any condition.